### PR TITLE
Rename Dataset to Table

### PR DESCRIPTION
**Pull request by mfiume**
* rename dataset to table
* rename schema to data_model
* rename objects to data
* other renaming
* introduce new /data endpoint

### CODE REVIEW

**Review by mcupak — APPROVED** [review]

**Comment by davidlougheed** [comment]
~~Any thoughts on my concerns in #18? This still applies with the rename; and the README mentions `/datasets/{id}` instead of `/dataset/{id}` in some places as well.~~

**Review by mcupak — CHANGES_REQUESTED** [review]
Needs rebasing.

**Comment by patmagee** [comment]
@mfiume @mcupak FYI the README.md needs to be updated. All of the documentation is outdated

File: spec/search-api.yaml [hunk 1]
```diff
@@ -12,94 +12,98 @@ info:
...
         '500':
           description: An unexpected error occurred
-  /dataset/{dataset_id}:
+  /table/{table_name}/info:
     get:
-      summary: Get a dataset
-      description: Returns the first page of the dataset identified by dataset_id.
-      operationId: getDataset
+      summary: Get a Table
+      description: Returns the Table identified by name.
```
Comment (jfuerth) [inline spec/search-api.yaml:32]: suggest: "Returns information about the Table identified by table_name"

File: spec/search-api.yaml [hunk 2]
```diff
@@ -12,94 +12,98 @@ info:
...
               schema:
-                $ref: "#/components/schemas/Dataset"
+                $ref: "#/components/schemas/Table"
         '500':
           description: An unexpected error occurred
+  /table/{table_name}/data:
+    get:
+      summary: Fetch data from a Table
+      description: Optional operation that returns data from a Table
+      operationId: data
```
Comment (jfuerth) [inline spec/search-api.yaml:56]: maybe get_table_data?

File: spec/search-api.yaml [hunk 3]
```diff
@@ -12,94 +12,98 @@ info:
...
+      responses:
```
Comment (jfuerth) [inline spec/search-api.yaml:57]: Add a 200 success response that returns TableData?

File: spec/search-api.yaml [hunk 4]
```diff
@@ -12,94 +12,98 @@ info:
...
+        '404':
+          description: This server does not implement the data operation
   /search:
     post:
-      summary: (Placeholder) Perform a search
-      description: Optional operation that accepts a query and returns a dataset
+      summary: Perform a search on Tables
+      description: Optional operation that accepts a Query and returns a Table
       operationId: search
       responses:
```
Comment (jfuerth) [inline spec/search-api.yaml:65]: Add a 200 success response that returns TableData?

File: spec/search-api.yaml [hunk 5]
```diff
@@ -12,94 +12,98 @@ info:
...
-        location:
-          type: string
-          description: Dataset absolute or relative location
-          format: uri
+          description: Table name
         description:
           type: string
-          description: Optional dataset description
-        schema:
+          description: Optional description of the Table
```
Comment (jfuerth) [inline spec/search-api.yaml:91]: Suggest: Human-readable description of the Table (Optional)

**Review by jfuerth — APPROVED** [review]
Looks good. Just a few suggestions on wording.

We should also
* update DATASET.md (rename to tables.md and edit content)
* update content of README.md to refer to "Tables" rather than "Datasets" and "Data Model" rather than "Schema"

Doesn't need to block this, but should be done soon after merging.

**Comment by patmagee** [comment]
@mcupak #24 should be merved over this PR. I addressed the issues outlined by @jfuerth and updated some of the documentation

**Pull request merged**